src/testing/alerts.rs

```diff
@@ -0,0 +1,525 @@
+            Recipient::Node(alerter_index),
+        );
+    test_case.run(own_index).await;
+}
+
+#[tokio::test]
+async fn ignores_wrong_alert() {
+    let n_members = NodeCount(7);
+    let own_index = NodeIndex(0);
+    let alerter_index = NodeIndex(1);
+    let forker = NodeIndex(6);
+    let mut test_case = TestCase::new(n_members);
+    let valid_unit = test_case.unchecked_signed_unit(alerter_index, 0, 0).await;
+    let wrong_fork_proof = (valid_unit.clone(), valid_unit);
+    let wrong_alert = test_case.alert(forker, wrong_fork_proof.clone());
+    let signed_wrong_alert = test_case
+        .unchecked_signed(wrong_alert.clone(), forker)
+        .await;
+    let signed_wrong_alert_hash = test_case
+        .indexed_unchecked_signed(Signable::hash(&wrong_alert), own_index)
+        .await;
+    test_case
+        .incoming_message(AlertMessage::ForkAlert(signed_wrong_alert))
+        .unexpected_notification(ForkingNotification::Forker(wrong_fork_proof));
+    for i in 1..n_members.0 {
+        test_case.unexpected_message(
+            AlertMessage::RmcMessage(
+                own_index,
+                RmcMessage::SignedHash(signed_wrong_alert_hash.clone()),
+            ),
+            Recipient::Node(NodeIndex(i)),
+        );
+    }
+    // We also make a proper alert to actually have something to wait for.
+    let fork_proof = test_case.fork_proof(forker, 0).await;
+    let alert = test_case.alert(alerter_index, fork_proof.clone());
+    let signed_alert = test_case
+        .unchecked_signed(alert.clone(), alerter_index)
+        .await;
+    test_case
+        .incoming_message(AlertMessage::ForkAlert(signed_alert))
+        .outgoing_notification(ForkingNotification::Forker(fork_proof));
+    test_case.run(own_index).await;
+}
+
+#[tokio::test]
+async fn responds_to_alert_queries() {
+    let n_members = NodeCount(7);
+    let own_index = NodeIndex(0);
+    let querier = NodeIndex(1);
+    let forker = NodeIndex(6);
+    let mut test_case = TestCase::new(n_members);
+    let alert = test_case.alert(own_index, test_case.fork_proof(forker, 0).await);
+    let alert_hash = Signable::hash(&alert);
+    let signed_alert = test_case.unchecked_signed(alert.clone(), own_index).await;
+    let signed_alert_hash = test_case
+        .indexed_unchecked_signed(alert_hash, own_index)
+        .await;
+    test_case
+        .incoming_alert(alert.clone())
+        .outgoing_message(
+            AlertMessage::ForkAlert(signed_alert.clone()),
+            Recipient::Everyone,
+        )
+        .outgoing_message(
+            AlertMessage::RmcMessage(own_index, RmcMessage::SignedHash(signed_alert_hash.clone())),
+            Recipient::Node(querier),
+        )
+        .wait()
+        .incoming_message(AlertMessage::AlertRequest(querier, alert_hash))
+        .outgoing_message(
+            AlertMessage::ForkAlert(signed_alert.clone()),
+            Recipient::Node(querier),
+        );
```

Comment:
In the loop belowe we test this once again. Is that intentional?

Reply:
Yup, we want a response to every query, including repeated ones.